drf_spectacular/plumbing.py

```diff
@@ -657,11 +657,15 @@ def modify_for_versioning(patterns, method, path, view, requested_version):
         except Resolver404:
             error(f"namespace versioning path resolution failed for {path}. path will be ignored.")
     elif issubclass(view.versioning_class, versioning.AcceptHeaderVersioning):
-        renderer = view.get_renderers()[0]
-        view.request.META['HTTP_ACCEPT'] = f'{renderer.media_type}; version={requested_version}'
-
-        negotiated = view.perform_content_negotiation(view.request)
-        view.request.accepted_renderer, view.request.accepted_media_type = negotiated
+        # Append the version into request accepted_media_type.
+        # e.g "application/json; version=1.0"
+        # To allow the AcceptHeaderVersioning negotiator going through.
+        if not hasattr(view.request, 'accepted_renderer'):
+            # Probably a mock request, content negotation was not performed  so we do it now.
+            negotiated = view.perform_content_negotiation(view.request)
+            view.request.accepted_renderer, view.request.accepted_media_type = negotiated
+        if '; version=' not in view.request.accepted_media_type:
+            view.request.accepted_media_type += f'; version={requested_version}'
```

Comment:
your change is absolutely correct. i would just prefer that setting `accepted_media_type` would not be conditional with an `if` and instead override it always. reason is this condition currently aligns perfectly with the version estimation here
https://github.com/tfranzel/drf-spectacular/blob/master/drf_spectacular/generators.py#L144
but if that changes this might break. always setting it would be more robust i believe.

i noticed one more thing. we should probably use `view.versioning_class.version_param` instead of `"version"` because it is configurable in DRF. that should fix these couple of lines for good then.

Reply:
> your change is absolutely correct. i would just prefer that setting `accepted_media_type` would not be conditional with an `if` and instead override it always. reason is this condition currently aligns perfectly with the version estimation here
> https://github.com/tfranzel/drf-spectacular/blob/master/drf_spectacular/generators.py#L144

the `if` statement is there to prevent the version to be appended again and again
```
application/json; version=1.0; version=1.0; version=1.0;
```

> but if that changes this might break. always setting it would be more robust i believe.
> 
> i noticed one more thing. we should probably use `view.versioning_class.version_param` instead of `"version"` because it is configurable in DRF. that should fix these couple of lines for good then.

makes perfect sense. Thanks, I will change it.

Reply:
We could also use `view.versioning_class.determine_version(request, **view.kwargs)`
in the if statement. Might better than checking for the raw string content `if '; version=' not in view.request.accepted_media_type`